packages/shared/src/contexts/BootProvider.tsx

```diff
@@ -146,6 +155,12 @@ export const BootDataProvider = ({
     },
   );
 
+  useEffect(() => {
+    if (error && !hasGlobalError) {
+      setGlobalError(true);
+    }
+  }, [error, hasGlobalError]);
+
```

Comment:
Could this not be just evaluation, if boot has error return `<ServerError />`?

Reply:
Yeah tested in this new flow we don't need state/effect :)